src/components/Profile/ProfileComponent.jsx

```diff
@@ -0,0 +1,168 @@
+// React Imports
+import React, { useContext, useEffect, useState } from 'react';
+// Custom Hook Imports
+import { useSession } from '@hooks';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import EditIcon from '@mui/icons-material/Edit';
+// Context Imports
+import { SignedInUserContext } from '@contexts';
+// Component Inputs
+import ProfileImageField from './ProfileImageField';
+import ProfileInputField from './ProfileInputField';
+
+/**
+ * @typedef {import("../../typedefs.js").profileComponentProps} profileComponentProps
+ */
+
+/**
+ * The UserProfile Component is a component that renders the user's profile on
+ * PASS
+ *
+ * @memberof Profile
+ * @name ProfileComponent
+ * @param {profileComponentProps} Props - Props for ClientProfile component
+ * @returns {React.JSX.Element} The UserProfile Component
+ */
+const ProfileComponent = ({ clientProfile }) => {
+  const { session } = useSession();
+  const { updateProfileInfo, setProfileData, profileData, fetchProfileInfo } =
+    useContext(SignedInUserContext);
+
+  const [profileName, setProfileName] = useState(profileData?.profileInfo.profileName);
+  const [nickname, setNickname] = useState(profileData?.profileInfo.nickname);
+  const [edit, setEdit] = useState(false);
+
+  const loadProfileData = async () => {
+    const profileDataSolid = await fetchProfileInfo(session);
+    setProfileData(profileDataSolid);
+
+    setProfileName(profileDataSolid.profileInfo.profileName);
+    setNickname(profileDataSolid.profileInfo.nickname);
+  };
+
+  const handleCancelEdit = () => {
+    loadProfileData();
+    setEdit(!edit);
+  };
+
+  const handleEditInput = () => {
+    setEdit(!edit);
+  };
+
+  const handleUpdateProfile = async (event) => {
+    event.preventDefault();
+
+    const inputValues = {
+      profileName,
+      nickname
+    };
+
+    await updateProfileInfo(session, profileData, inputValues);
+
+    loadProfileData();
+    setEdit(false);
+  };
+
+  useEffect(() => {
+    loadProfileData();
+  }, []);
+
+  return (
+    <Box
+      style={{
+        display: 'flex',
+        gap: '15px',
+        padding: '10px'
+      }}
+    >
+      <ProfileImageField loadProfileData={loadProfileData} clientProfile={clientProfile} />
+      <form
+        onSubmit={handleUpdateProfile}
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          boxShadow: '0 0 3px 0 black',
+          justifyContent: 'space-between',
+          padding: '20px'
+        }}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            gap: '10px'
+          }}
+        >
+          <>
```

Comment:
don't think this fragment is needed/wanted here

Reply:
Good catch! I'll have these removed.